Fix/restore the display of error messages in comment form jQuery validation

calumryan opened the pull request:
Fixes to jQuery custom plugin validation to restore the display of error messages when submitting the comments form with empty fields. 

- Fixed button selector to class name (previously targetting a non-existent submit input)
- Restored missing `<header>` landmark required for error list output
- Changed class names based on design system error summary example https://design-system.service.gov.uk/components/error-summary/

![Screenshot of comment form with error messages](https://user-images.githubusercontent.com/2226904/107783248-f4a9c280-6d41-11eb-94b8-3c600649284e.png)

RobjS commented:
@calumryan This looks good, well done for spotting it needed fixing!

It might be useful to have some of the explanatory text from the PR in the commit message, so it's available locally or if the PR itself is no longer available. And it'd be good to update the "Unreleased" section of the `CHANGELOG.md` with a line on this, so we know what's included in the next release.

calumryan commented:
> It might be useful to have some of the explanatory text from the PR in the commit message, so it's available locally or if the PR itself is no longer available. And it'd be good to update the "Unreleased" section of the `CHANGELOG.md` with a line on this, so we know what's included in the next release.

@RobjS Thanks, I've renamed this and updated the CHANGELOG 👍🏻
